src/main/java/jchess/ruleengine/DefaultRuleProcessor.java

```diff
@@ -432,4 +432,97 @@ public IPlayerAgent tryCheckIfPlayerEndengered(IBoardAgent oBoard, IPlayerAgent
 				return null;	
 	}
 
+	/**
+	 * Checking for Stalemate Rule
+	 * 
+	 * Stalemate is a situation in the game of chess where the player whose turn 
+	 * it is to move is not in check but has no legal move. 
+	 * The rules of chess provide that when stalemate occurs, the game ends as a draw.
+	 */
+	
+	public Boolean checkStalemate(IBoardAgent oBoard, IPlayerAgent oPlayer) {
+		Boolean bStalemateCheckIfPlayerEndengered = false;
+		Boolean bStalemateCheckIfPlayNotPossible = false;
+		int nCountNoOfPieces =0;
+		int nCountPiecesTrapped=0;
+		int nCounterOfKingMoves=0;
+		int[] iaArrayOfPossibleChecks = new int[]{0,0,0,0,0,0,0,0,0}; 
```

Comment:
Could you please tell me what this array is taken for? It would be helpful if you write your logic that how you achieved the goal.

Reply:
this array is taken for the movements that the king can make (max movement is 8) and we find out the possible movement that can lead to checkmate. If all the movements of the king leads to checkmate and the current position is not a check and no other pieces of the current player can be moved then its a stalemate resulting in a draw